backend-lib/src/main/rust/lib.rs

```diff
@@ -438,17 +421,128 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_createAcc
-            .create_account(&seed, &birthday)
+        let account_name = java_string_to_rust(env, &account_name);
+        let key_source = java_nullable_string_to_rust(env, &key_source);
+
+        let (account_uuid, usk) = db_data
+            .create_account(
+                &account_name,
+                &seed,
+                &birthday,
+                key_source.as_ref().map(|s| s.as_ref()),
+            )
             .map_err(|e| anyhow!("Error while initializing accounts: {}", e))?;
 
-        let account = db_data.get_account(account_id)?.expect("just created");
-        let account_index = match account.source() {
-            AccountSource::Derived { account_index, .. } => account_index,
-            AccountSource::Imported { .. } => unreachable!("just created"),
-        };
+        Ok(encode_usk(env, account_uuid, usk)?.into_raw())
+    });
+    unwrap_exc_or(&mut env, res, ptr::null_mut())
+}
 
-        Ok(encode_usk(env, account_index, usk)?.into_raw())
+/// Tells the wallet to track an account using a unified full viewing key.
+///
+/// Returns details about the imported account, including the unique account identifier for
+/// the newly-created wallet database entry. Unlike the other account creation APIs
+/// ([`Self::create_account`] and [`Self::import_account_hd`]), no spending key is returned
+/// because the wallet has no information about how the UFVK was derived.
+///
+/// Certain optimizations are possible for accounts which will never be used to spend funds. If
+/// `spending_key_available` is `false`, the wallet may choose to optimize for this case, in
+/// which case any attempt to spend funds from the account will result in an error.
+///
+/// The [`WalletWrite`] trait documentation has more details about account creation and import.
+///
+/// # Arguments
+/// - `account_name`: A human-readable name for the account.
+/// - `ufvk_str`: The UFVK used to detect transactions involving the account.
+/// - `purpose`: Metadata describing whether or not data required for spending should be
+///   tracked by the wallet.
+/// - `treestate`: The tree state corresponding to the last block prior to the wallet's
+///   birthday height.
+/// - `recover_until`: An optional height at which the wallet should exit "recovery mode". In
+///    order to avoid confusing shifts in wallet balance and spendability that may temporarily be
+///    visible to a user during the process of recovering from seed, wallets may optionally set a
+///    "recover until" height. The wallet is considered to be in "recovery mode" until there
+///    exist no unscanned ranges between the wallet's birthday height and the provided
+///   `recover_until` height, exclusive.
+/// - `key_source`: A string identifier or other metadata describing the source of the seed.
+///   This is treated as opaque metadata by the wallet backend; it is provided for use by
+///   applications which need to track additional identifying information for an account.
```

Comment:
```suggestion
/// - `key_source`: A string identifier or other metadata describing the source of the seed.
///   This is treated as opaque metadata by the wallet backend; it is provided for use by
///   applications which need to track additional identifying information for an account.
/// - `ufvk_str`: The UFVK used to detect transactions involving the account.
/// - `treestate`: The tree state corresponding to the last block prior to the wallet's
///   birthday height.
/// - `recover_until`: An optional height at which the wallet should exit "recovery mode". In
///    order to avoid confusing shifts in wallet balance and spendability that may temporarily be
///    visible to a user during the process of recovering from seed, wallets may optionally set a
///    "recover until" height. The wallet is considered to be in "recovery mode" until there
///    exist no unscanned ranges between the wallet's birthday height and the provided
///   `recover_until` height, exclusive.
/// - `purpose`: 0 (Spending) if data required for spending should be tracked by the wallet,
///   or 1 (ViewOnly) if the account will never be used to spend funds.
/// - `seed_fingerprint_bytes`: the [seed fingerprint]. Only used if `purpose` is 0 (Spending).
/// - `hd_account_index_raw`: the ZIP 32 account index. Only used if `purpose` is 0 (Spending).
///
/// [seed fingerprint]: https://zips.z.cash/zip-0032#seed-fingerprints
```